W-23259833-maskSecureProps-RTM-dm (#898)
mulesoft/docs-runtime-manager: cloudhub/modules/ROOT/pages/secure-application-properties.adoc

For applications deployed to CloudHub workers or hybrid deployments, you can use Runtime Manager to set hidden properties.
For applications that you deploy using another deployment option, you must use secure property placeholder files bundled in the deployable application archive file.

[NOTE]
====
This page describes safely hidden application properties for CloudHub and hybrid deployments.

> On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection). Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security secrets manager. Runtime Manager properties override archive-bundled properties of the same name, so masking values through `mule-artifact.json` or a configuration file doesn't take effect on CloudHub 2.0. For more information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].

Reviewer: ```suggestion
On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection). Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security Secrets Manager. Runtime Manager properties override archive-bundled properties of the same name, so configuring value masking through `mule-artifact.json` or a configuration file doesn't take effect on CloudHub 2.0. For more information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].
```

> On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection). Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security Secrets Manager. Runtime Manager properties override archive-bundled properties of the same name, so configuring value masking through `mule-artifact.json` or a configuration file doesn't take effect on CloudHub 2.0. For more information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].

Author: ```suggestion
On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection). Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security secrets manager. Runtime Manager properties override archive-bundled properties of the same name, so configuring value masking through `mule-artifact.json` or a configuration file doesn't take effect on CloudHub 2.0. For more information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].
```

Author: as per the product docs, "Anypoint Security secrets manager"

For applications deployed to CloudHub workers or hybrid deployments, you can use Runtime Manager to set hidden properties.
For applications that you deploy using another deployment option, you must use secure property placeholder files bundled in the deployable application archive file.

[NOTE]
====

> This page describes safely hidden application properties for CloudHub (1.0) and hybrid deployments.

Author: ```suggestion
This page describes safely hidden application properties for CloudHub and hybrid deployments.
```

On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection), not through `mule-artifact.json` or config-file masking. Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security secrets manager. Runtime Manager properties override archive-bundled properties of the same name.

> On CloudHub 2.0, configuring masking through `mule-artifact.json` or config-file masking isn't the recommended practice. For information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].

Author: ```suggestion
On CloudHub 2.0, secure-property masking is configured in Runtime Manager on the *Properties* tab (property protection). Masked values aren't viewable or retrievable after being set and are encrypted in the Anypoint Security secrets manager. Runtime Manager properties override archive-bundled properties of the same name, so masking values through `mule-artifact.json` or a configuration file doesn't take effect on CloudHub 2.0. For more information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].
```

> On CloudHub 2.0, configuring masking through `mule-artifact.json` or config-file masking isn't the recommended practice. For information, see xref:cloudhub-2::ch2-protect-app-props.adoc[Protecting App Property Values].

Author: ```suggestion
On CloudHub 2.0, configuring masking through `mule-artifact.json` or config-file masking isn't the recommended practice. For information, see xref:cloudhub-2::ch2-protect-app-props.adoc[].
```

== See Also

* xref:mule-runtime::secure-configuration-properties.adoc[Secure Configuration Properties]

> * xref:cloudhub-2::ch2-protect-app-props.adoc[Protecting App Property Values (CloudHub 2.0)]

Author: ```suggestion
* xref:cloudhub-2::ch2-protect-app-props.adoc[Protecting App Property Values on CloudHub 2.0]
```
